File: python/scripts/get_board.py
#!/usr/bin/env python
import argparse
import sys
from os.path import abspath, dirname, join

sys.path.append(abspath(join(dirname(__file__), "..", "src")))


from access_token import AccessToken
from api_config import ApiConfig
from arguments import common_arguments
from board import Board
from oauth_scope import Scope
from pin import Pin


def main(argv=[]):
    """
    This script prints the information associated with a board. The board identifier
    may be obtained with the get_user_boards script. If the board has any sections,
    the script will print information about them.

    Specify the --pins argument to print information about each pin on the board
    and its sections.
    """
    parser = argparse.ArgumentParser(description="Get a Board")
    parser.add_argument("-b", "--board-id", required=True, help="board identifier")
    parser.add_argument(
        "--pins", action="store_true", help="Get the Pins for the Board"
    )
    common_arguments(parser)
    args = parser.parse_args(argv)

    # get configuration from defaults and/or the environment
    api_config = ApiConfig(verbosity=args.log_level)

    # Note: It's possible to use the same API configuration with
    # multiple access tokens, so these objects are kept separate.
    access_token = AccessToken(api_config, name=args.access_token)
    scopes = [Scope.READ_USERS, Scope.READ_BOARDS]
    if args.pins:
        scopes.append(Scope.READ_PINS)
    access_token.fetch(scopes=scopes)

    board = Board(args.board_id, api_config, access_token)
    board_data = board.get()
    board.print_summary(board_data)

    if args.pins:
        for pin_data in board.get_pins():
            # ignore pins in sections for now. they will be printed for each section
            if not pin_data.get("board_section_id"):
                Pin.print_summary(pin_data)

    for section_data in board.get_sections():
        board.print_section(section_data)
        if args.pins:
            for pin_data in board.get_section_pins(section_data["id"]):
                Pin.print_summary(pin_data)


if __name__ == "__main__":
    main(sys.argv[1:])

File: nodejs/scripts/get_board.js
#!/usr/bin/env node
import { ArgumentParser } from 'argparse';

import { AccessToken } from '../src/access_token.js';
import { ApiConfig } from '../src/api_config.js';
import { Board } from '../src/board.js';
import { common_arguments } from '../src/arguments.js';
import { Pin } from '../src/pin.js';
import { Scope } from '../src/oauth_scope.js';

/**
 * This script prints the information associated with a pin. The pin identifier
 * my be obtained with the get_user_pins.py or get_board.py script.
 */
async function main(argv) {
  const parser = new ArgumentParser({
    description: 'Get A Board'
  });
  parser.add_argument('-b', '--board-id', { required: true, help: 'board identifier' });
  parser.add_argument('--pins', { action: 'store_true', help: 'Get the Pins for the Board' });
  common_arguments(parser);
  const args = parser.parse_args(argv);

  // get configuration from defaults and/or the environment
  const api_config = new ApiConfig({ verbosity: args.log_level });

  // Note: It's possible to use the same API configuration with
  // multiple access tokens, so these objects are kept separate.
  const access_token = new AccessToken(api_config, { name: args.access_token });
  const scopes = [Scope.READ_USERS, Scope.READ_BOARDS];
  if (args.pins) {
    scopes.push(Scope.READ_PINS);
  }
  await access_token.fetch({ scopes: scopes });

  const board = new Board(args.board_id, api_config, access_token);
  const board_data = await board.get();
  Board.print_summary(board_data);

  if (args.pins) {
    const pin_iterator = await board.get_pins();
    for await (const pin_data of pin_iterator) {
      // ignore pins in sections for now. they will be printed for each section
      if (pin_data.board_section_id) {
        continue;
      }
      Pin.print_summary(pin_data);
    }
  }

  const section_iterator = await board.get_sections();
  for await (const section_data of section_iterator) {
    Board.print_section(section_data);
    if (args.pins) {
      const pin_iterator = await board.get_section_pins(section_data.id);
      for await (const pin_data of pin_iterator) {
        Pin.print_summary(pin_data);
      }
    }
  }
}

if (!process.env.TEST_ENV) {
  main(process.argv.slice(2));
}
